src/views/citation_editable.tsx

```diff
@@ -239,17 +220,14 @@ export class CitationEditableView extends CitationView<
         item._id = existingItem._id
       } else {
         triggerUpdate = true
-        // add the item to the model map so it's definitely available
-        setLibraryItem(item as BibliographyItem)
```

Comment:
So you are not updating the library while you still depend on it in L215 to `matchLibraryItemByIdentifier`.
This will result in inserting the node every time because it does not exist in the library. 

To produce it:
- Cite a newly added bibliography item. 
- Approve the changes and complete the task -> It will be added to the references section.
- Try to cite (somewhere else) that bibliography item again -> It will add it again to the references section.

Reply:
updated, to include elements in the documents